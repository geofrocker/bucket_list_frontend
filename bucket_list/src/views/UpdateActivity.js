import React, {Component} from 'react';
import Header from '../components/Header';
import Sidebar from '../components/Sidebar';
import axios from 'axios';
import {Redirect} from 'react-router-dom';
import swal from 'sweetalert';

class UpdateActivity extends Component{
    constructor(props){
        super(props);
        let bucket_id = this.props.match.params.bucket_id;
        let item_id = this.props.match.params.item_id;
        this.state = {description: '', bucket_id: bucket_id, item_id:item_id};
        this.handleDescriptionChange = this.handleDescriptionChange.bind(this);
        this.handleUpdateData = this.handleUpdateData.bind(this);
    }

    handleDescriptionChange(event){
        this.setState({description: event.target.value})
    }

    componentDidMount() {
        axios({
            url: 'http://127.0.0.1:5000/api/v1/callback',
            method:"GET",
            headers: {
                'token': window.localStorage.getItem('token'),
                'Content-Type': 'application/json'
            }
        }).then((response)=>{
            this.setState({isAuthorized: true});
            window.localStorage.setItem('isLoggedIn', true)
        }).catch((xhr)=>{
            this.setState({isAuthorized: false});
            window.localStorage.setItem('isLoggedIn', false)
        });

        let url = "http://127.0.0.1:5000/api/v1/bucketlists/" + this.state.bucket_id + "/items/" + this.state.item_id;

        axios({

            url: url,
            method: "GET",
            headers: {
                'token': window.localStorage.getItem('token'),
                'Content-Type': 'application/json'
            }
        })
            .then((response)=>{
                let description = response.data.activity.description;
                this.setState ({description:description})
            })
            .catch((xhr) =>{
                swal("Error!", xhr.response.data.error, "error");
                if (xhr.response.status == 404){
                    this.setState({redirect:true})
                }

            });
    }

    handleUpdateData(event){
        event.preventDefault();
        let bucket_id = this.state.bucket_id;
        let item_id = this.state.item_id;
        let url = "http://127.0.0.1:5000/api/v1/bucketlists/" + this.state.bucket_id + "/items/" + this.state.item_id;
        let data = {description: this.state.description};
        axios({
            url: url,
            method: 'PUT',
            data: data,
            datatype: 'json',
            headers: {
                'token': window.localStorage.getItem('token'),
                'Content-Type': 'application/json'
            },
        })
            .then((response)=>{
                this.setState({redirect:true});
            })
            .catch((xhr) =>{
                swal("Error!", xhr.response.data.error, "error");


            });

    }

    render(){
        if (this.state.redirect){
            return <Redirect to={"/bucketlists/" + this.state.bucket_id + "/items/view/"}/>
        }

        if (!this.state.isAuthorized){
            return(
                <div>
                    <article className="content item-editor-page">
                        <div className="card card-block">
                            <p>Unauthorized! Please log in</p>
                        </div>
                    </article>
                </div>
            )
        }


        return(
            <div>
                <Header/>
                <Sidebar/>
                <article className="content item-editor-page">
                    <div className="title-block">
                        <h3 className="title"> Update Activity
                            <span className="sparkline bar"></span>
                        </h3>
                    </div>

                    <form method="post" onSubmit={this.handleUpdateData}>
                        <div className="card card-block">
                            <div className="form-group row">
                                <label className="col-sm-2 form-control-label text-xs-right">
                                    Name:
                                </label>

                                <div className="col-sm-10">
                                    <input type="text" name="description" value={this.state.description} onChange={this.handleDescriptionChange} className="form-control boxed" placeholder="" />
                                </div>
                            </div>


                            <div className="col-sm-10 col-sm-offset-2">
                                <button type="submit" className="btn btn-primary">
                                    Update
                                </button>
                            </div>
                        </div>
                    </form>
                </article>
            </div>
        )

    }
}
export default UpdateActivity;