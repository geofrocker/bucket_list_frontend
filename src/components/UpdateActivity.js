import React, {Component} from 'react';
import Header from './Header';
import Sidebar from './Sidebar';
import axios from 'axios';
import {Redirect} from 'react-router-dom';
import swal from 'sweetalert';

class UpdateActivity extends Component{
    /**
     * This component is used to create activities and items.
     * It handles the following methods:
     *  handleDescriptionChange --> used to set the description state in the constructor
     *  handleUpdateData --> Used to post data in the end point
     *  render --> Used to return jsx
     */

    constructor(props){
        super(props);
        let bucket_id = this.props.match.params.bucket_id;
        let item_id = this.props.match.params.item_id;
        this.state = {description: '', bucket_id: bucket_id, item_id:item_id};
        this.handleDescriptionChange = this.handleDescriptionChange.bind(this);
        this.handleUpdateData = this.handleUpdateData.bind(this);
    }

    handleDescriptionChange(event){
        this.setState({description: event.target.value})
    }

    componentDidMount() {
        axios({
            url: 'https://ridge-bucket-list-api.herokuapp.com/api/v1/callback',
            method:"GET",
            headers: {
                'token': window.localStorage.getItem('token'),
                'Content-Type': 'application/json'
            }
        }).then(()=>{
            this.setState({isAuthorized: true});
            window.localStorage.setItem('isLoggedIn', true)
        }).catch(()=>{
            this.setState({isAuthorized: false, login_redirect:true});
        });

        let url = "https://ridge-bucket-list-api.herokuapp.com/api/v1/bucketlists/" + this.state.bucket_id + "/items/" + this.state.item_id;

        axios({

            url: url,
            method: "GET",
            headers: {
                'token': window.localStorage.getItem('token'),
                'Content-Type': 'application/json'
            }
        })
            .then((response)=>{
                let description = response.data.activity.description;
                this.setState ({description:description})
            })
            .catch((xhr) =>{
                swal("Error!", xhr.response.data.error, "error");
                if (xhr.response.status === 404){
                    this.setState({redirect:true})
                }

            });
    }

    handleUpdateData(event){
        event.preventDefault();
        let bucket_id = this.state.bucket_id;
        let item_id = this.state.item_id;
        let url = "https://ridge-bucket-list-api.herokuapp.com/api/v1/bucketlists/" + bucket_id + "/items/" + item_id;
        let data = {description: this.state.description};
        axios({
            url: url,
            method: 'PUT',
            data: data,
            datatype: 'json',
            headers: {
                'token': window.localStorage.getItem('token'),
                'Content-Type': 'application/json'
            },
        })
            .then(()=>{
                this.setState({redirect:true});
            })
            .catch((xhr) =>{
                swal("Error!", xhr.response.data.error, "error");


            });

    }

    render(){
        if(this.state.login_redirect){
            return(<Redirect to={'/login/'}/>)
        }

        if (this.state.redirect){
            return <Redirect to={"/bucketlists/" + this.state.bucket_id + "/items/view/"}/>
        }

        if (!this.state.isAuthorized){
            return(
                <div>
                    <article className="content item-editor-page">

                    </article>
                </div>
            )
        }


        return(
            <div>
                <Header/>
                <Sidebar/>
                <article className="content item-editor-page">
                    <div className="title-block">
                        <h3 className="title"> Update Activity
                            <span className="sparkline bar"></span>
                        </h3>
                    </div>

                    <form method="post" id="update_activity_form" onSubmit={this.handleUpdateData}>
                        <div className="card card-block">
                            <div className="form-group row">
                                <label className="col-sm-2 form-control-label text-xs-right">
                                    Name:
                                </label>

                                <div className="col-sm-10">
                                    <input type="text" name="description" value={this.state.description} onChange={this.handleDescriptionChange} className="form-control boxed" placeholder="" />
                                </div>
                            </div>


                            <div className="col-sm-10 col-sm-offset-2">
                                <button type="submit" className="btn btn-primary">
                                    Update
                                </button>
                            </div>
                        </div>
                    </form>
                </article>
            </div>
        )

    }
}
export default UpdateActivity;