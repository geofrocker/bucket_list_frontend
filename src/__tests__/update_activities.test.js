import React from 'react';
import UpdateActivity from '../views/UpdateActivity';

import {mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import {MemoryRouter} from 'react-router-dom';

describe('<UpdateBucket />', ()=>{
    beforeEach(() => {
        global.localStorage = {
            token: 'someToken',
            getItem: function () {
                return 'someToken'
            }
        };
    });

    it('contains update activity form', function () {
        const props = {
            match:{
                params: {bucket_id:1}
            }
        };
        const wrapper = mount(
            <MemoryRouter>
                <UpdateActivity {...props}/>
            </MemoryRouter>);

        const preventDefault = jest.fn();
        wrapper.setState({description:"test description", isAuthorized:true});

        setTimeout(() => {
            expect(wrapper.find("#update_activity_form").length === 1);
            wrapper.find("#update_activity_form").simulate('submit', {preventDefault})

        }, 60);
        expect(toJson(wrapper)).toMatchSnapshot();


    });

    it('should change values in the form', function () {
        const props = {
            match:{
                params: {bucket_id:1, item_id:1}
            }
        };
        const wrapper = mount(
            <MemoryRouter>
                <UpdateActivity {...props}/>
            </MemoryRouter>);
        setTimeout(() => {
            wrapper.find("description").simulate('change', {target: {value:'test description'}});
        }, 60);
        expect(toJson(wrapper)).toMatchSnapshot();
    })



});

