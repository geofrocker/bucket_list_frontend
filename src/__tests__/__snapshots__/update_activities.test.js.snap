// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpdateBucket /> contains update activity form 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "mipru8",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "mipru8",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <UpdateActivity
      match={
        Object {
          "params": Object {
            "bucket_id": 1,
          },
        }
      }
    >
      <div>
        <article
          className="content item-editor-page"
        />
      </div>
    </UpdateActivity>
  </Router>
</MemoryRouter>
`;

exports[`<UpdateBucket /> should change values in the form 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "zawlli",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "zawlli",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <UpdateActivity
      match={
        Object {
          "params": Object {
            "bucket_id": 1,
            "item_id": 1,
          },
        }
      }
    >
      <div>
        <article
          className="content item-editor-page"
        />
      </div>
    </UpdateActivity>
  </Router>
</MemoryRouter>
`;
